### ISSUE TITLE
Smart pointer documentation is out of date

### ISSUE THREAD

**Issue by nico** [SerenityOS/serenity#17432]
Documentation/SmartPointers.md mentions `make<>()`, `adopt_own()`, `adopt_own_if_nonnull()`, `make_ref_counted<>(), `tadopt_ref()`, `adopt_ref_if_nonnull<>()`, but it does _not_ mention `adopt_nonnull_own_or_enomem` nor`adopt_nonnull_ref_or_enomem()`.

(It _does_ mention `try_make<>()` and `try_make_ref_counted<>()`, which wrap the `_or_enomem()` functions.)

It also doesn't mention when to use which variant. (I'm guessing newer code should use the enomem variant.)

Separately, Documentation/Patterns.md has a section on "Fallible Constructors" which _does_ mention `adopt_nonnull_own_or_enomem` but not the more convenient `try_make<>()` wrapper around it. (It also doesn't mention refptrs.)

Suggestions:

1. Update SmartPointers.md to mention the _enomen functions, and rewrite it a bit to say "new code should use the ErrorOr<> versions, see also 'Fallible Constructors' Patterns.md"
2. The section in Patterns.md should use try_make instead of what it currently uses

(I'm not sure those are the right suggestions, though.)

@awesomekling @BertalanD @timschumi @IdanHo

**Comment by Nishchay1571999** [comment]
Is this issue open? If yes i would like to take it up

**Comment by GMTA** [comment]
@Nishchay1571999 yes it's still open, but it's probably better if it's picked up by someone with extensive experience in both the project and the nuances of our smart pointers. But thanks anyway!

**Comment by Nishchay1571999** [comment]
@GMTA hi thanks for the reply. I thought it was for begginers. Thats why i asked it to be assigned to me. Thats ok if its a complex. I would have required extensive help anyway.😅
